src/idpi/operators/support_gis.py

```diff
@@ -0,0 +1,229 @@
+"""Support geospatial logic."""
+
+# Standard library
+import dataclasses as dc
+import typing
+from collections.abc import Mapping
+
+# Third-party
+import numpy as np
+import xarray as xr
+
+# Local
+from .support_operators import get_grid_coords
+
+
+@dc.dataclass
+class RotLatLonGrid:
+    """Class representing a rotated lat lon grid.
+
+    Attributes
+    ----------
+    rlon : xr.DataArray
+        longitude values in degrees of the grid points in the rotated lat lon CRS.
+    rlat : xr.DataArray
+        latitude values in degrees of the grid points in the rotated lat lon CRS.
+    north_pole_lon : float
+        longitude of the rotated north pole in degrees.
+    north_pole_lat : float
+        latitude of the rotated north pole in degrees.
+
+    """
+
+    # all units in degrees
+    rlon: xr.DataArray
+    rlat: xr.DataArray
+    north_pole_lon: float
+    north_pole_lat: float
+
+
+def _check_requirements(geo: Mapping[str, typing.Any]) -> None:
+    requirements = {
+        "angleOfRotationInDegrees": 0.0,
+        "gridType": "rotated_ll",
+        "iScansNegatively": 0,
+        "jScansPositively": 1,
+    }
+    errors = {key: geo[key] for key in requirements if requirements[key] != geo[key]}
+    if errors:
+        msg = f"Unsupported values for keys: {errors}"
+        raise RuntimeError(msg)
+
+
+def get_grid(geo: Mapping[str, typing.Any]) -> RotLatLonGrid:
+    """Get grid parameters for a given field.
```

Comment:
Docstring should maybe state that this function gets grid parameters for a given field _defined on rot lat lon grid_, since it raises error otherwise?